packages/manager/modules/exchange/src/domain/dkim-autoconfig/domain-dkim-autoconfig.controller.js

```diff
@@ -0,0 +1,110 @@
+import { DKIM_CONFIGURATION_GUIDE } from './domain-dkim-autoconfig.constants';
+
+export default class ExchangeDomainDkimAutoconfigCtrl {
+  /* @ngInject */
+  constructor(
+    $q,
+    $scope,
+    wucExchange,
+    ExchangeDomains,
+    navigation,
+    messaging,
+    $translate,
+    exchangeStates,
+    coreConfig,
+  ) {
+    this.services = {
+      $q,
+      $scope,
+      wucExchange,
+      ExchangeDomains,
+      navigation,
+      messaging,
+      $translate,
+      exchangeStates,
+    };
+
+    this.$routerParams = wucExchange.getParams();
+    this.domain = navigation.currentActionData.domain;
+    this.dkimStatus = navigation.currentActionData.dkimStatus;
+    this.GLOBAL_DKIM_STATUS =
+      navigation.currentActionData.constant.GLOBAL_DKIM_STATUS;
+
+    this.dkimGuideLink =
+      DKIM_CONFIGURATION_GUIDE[coreConfig.getUser().ovhSubsidiary] ||
+      DKIM_CONFIGURATION_GUIDE.DEFAULT;
+
+    this.loading = true;
+
+    this.services.ExchangeDomains.gettingDNSSettings(
```

Comment:
to move to `$onInit` function

Reply:
I'm at a place where we have legacy code.
The data-wizard is built during the constructor and the $onInit is played after it has been built, so button management is not good if I put it in an onInit (I've lost some hair :blush: ).